### PR TITLE
atom: use exports over globals.

### PR DESCRIPTION
**Pull request by GlenCFL**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [Atom Public API](https://atom.io/docs/api/v1.21.0)
[Atom Exports](https://github.com/atom/atom/tree/v1.21.0/exports)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

This converts `@types/atom` into an actual external module, rather than the external/global namespace hybrid that it was previously. The idea is to begin to model the API itself, while sparing users the details of the underlying structure. Usage of the definitions is now consistent with typical ES6/TypeScript code, with module augmentation also working exactly the same way for every single type used within the API.

The connections between `@types/atom` and its submodules have been severed. This link has proven to be too restrictive both ways, as the usage cases for both are very different.  We have now properly defined the API for Atom, without making the definitions for the NPM modules themselves too restrictive for practical use. With the restriction lifted, a lot of vital functionality will be added to those NPM definitions soon, particularly to `@types/text-buffer`.

### CODE REVIEW

**Comment by GlenCFL** [comment]
The definitions for [event-kit](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/event-kit), [first-mate](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/first-mate), and [pathwatcher](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/pathwatcher) are now simple external modules as well. There isn't a strict API to abide by with these, so they depend on each other as usual.

These are all definitions where I am marked as the only owner. Due to this, these revamps would take ~20-30 days if I were to do this in multiple phases (which would be necessary), so everything is just tacked onto this one PR. This should be the last PR that targets all of these definitions, with each one from here on just targeting a single definition now that exports are revamped.

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Review by smhxx — APPROVED** [review]
Seems like solid housekeeping, and reduces global scope pollution to just the single `atom` global actually exposed by the API. LGTM. 👍

**Pull request merged**
